### PR TITLE
Service id update

### LINKED ISSUE #37: Service_id mandatory
Service_id is now mandatory but the NodeJS plugin does not provide a way to pass this param.

### PR DESCRIPTION
**Pull request by avilatusell**
Closes #37

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,8 +102,8 @@ Note that the **report** method supports sending the usage for multiple transact
 
 ```javascript
 var trans = [
-              { "app_id": "your application id", "usage": {"hits": 1}},
-              { "app_id": "your application id", "usage": {"hits": 1000}}
+              { "service_id": "your service id", "app_id": "your application id", "usage": {"hits": 1} },
```
Comment (davidor) [inline README.md]: This is not correct. The service id should not be included in each transaction.
Take a look at how `report` is defined: `report: (service_id, trans, callback)`

File: README.md [hunk 2]
```diff
@@ -63,10 +65,10 @@ var Client = require('3scale').Client;
 
 client = new Client("your provider key");
 
-client.authorize({ "app_id": "your application id", "app_key": "your application key" }, function(response){
+client.authorize({ "service_id": "your service id", "app_id": "your application id", "app_key": "your application key" }, function(response){
   if (response.is_success()) {
     var trans = [{ "app_id": "your application id", "usage": { "hits": 3 } }];
-    client.report(trans, function (response) {
+    client.report(service_id, trans, function (response) {
```
Comment (davidor) [inline README.md]: `service_id` should be `"your service id"` to be consistent with the `client.authorize` call. There are several examples like this.

File: test/integration_test.coffee [hunk 3]
```diff
@@ -18,7 +18,7 @@ describe 'Integration tests for the 3Scale::Client', ->
   describe 'The authorize method', ->
     it 'should call the callback with a successful response', (done) ->
       client = new Client provider_key
-      client.authorize {app_key: application_key, app_id: application_id}, (response) ->
+      client.authorize {service_id: service_id, app_key: application_key, app_id: application_id}, (response) ->
```
Comment (davidor) [inline test/integration_test.coffee]: I think you should also add the `service_id` in the other calls below too.  There's another `authorize` and a `report` in this test file.

File: test/client_test.coffee [hunk 4]
```diff
@@ -67,11 +67,11 @@ describe 'Basic test for the 3Scale::Client', ->
     it 'should call the callback with a successful response', (done) ->
       nock('https://su1.3scale.net')
         .get('/transactions/authorize.xml')
-        .query({ app_key: 'bar', app_id: 'foo', provider_key: '1234abcd' })
+        .query({ service_id: '1234567890987', app_key: 'bar', app_id: 'foo', provider_key: '1234abcd' })
         .reply(200, '<status><authorized>true</authorized><plan>Basic</plan></status>')
 
       client = new Client '1234abcd'
-      client.authorize {app_key: 'bar', app_id: 'foo'}, (response) ->
+      client.authorize {service_id: '1234567890987', app_key: 'bar', app_id: 'foo'}, (response) ->
```
Comment (davidor) [inline test/client_test.coffee]: You added the service id in two of the calls of this test file, but there are more. For example, the authorize call below this one (line 82).

Comment (avilatusell) [inline test/client_test.coffee]: I'll look into it. Thanks!

File: test/integration_test.coffee [hunk 5]
```diff
@@ -5,27 +5,28 @@ assert = require 'assert'
 provider_key = process.env.TEST_3SCALE_PROVIDER_KEY
 application_key = process.env.TEST_3SCALE_APP_KEY
 application_id = process.env.TEST_3SCALE_APP_ID
+service_id = process.env.TEST_3SCALE_SERVICE_ID
```
Comment (davidor) [inline test/integration_test.coffee]: This should be set in the `.travis.yml` file with the rest of envs: https://github.com/3scale/3scale_ws_api_for_nodejs/blob/master/.travis.yml

It should be encrypted like the provider_key: https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml

File: test/client_test.coffee (hunk 4 above)
Comment (davidor) [inline test/client_test.coffee]: You forgot the call in line 120:
`client.authorize { app_id: 'foo' }, (response) ->`

File: src/client.coffee [hunk 6]
```diff
@@ -269,7 +282,7 @@ module.exports = class Client
     Report transaction(s).
 
     Parameters:
-      service_id {String} Optional (required only if you have more than one service)
+      service_id {String} Required (from November 2016)
```
Comment (davidor) [inline src/client.coffee]: You forgot the service id in the example below: `client.report trans, (response) ->`

File: .travis.yml [hunk 7]
```diff
@@ -2,20 +2,22 @@ language: node_js
...
   global:
-    - CXX=g++-4.8
-    - TEST_3SCALE_APP_ID=4d4b20b9
-    - TEST_3SCALE_APP_KEY=ecce202ecc2eb8dc7a499c34a34d5987
-    - secure: B/3oWYZZttrXVM5IFn2evSuWMOQqEOOMol3nEQ2vEDGQ6rdGjpYLFDO/qMDPA8Af6Vt72lOFKkgzivuBp8nSd4S1YRTazilMle4zCJiJxzPs69kRDVXf/Mn8Qr/YrmgVzhblvdNqbKxTGrYyvDziPkliQKtIhqyILm3d6+CqmYw=
+  - CXX=g++-4.8
+  - TEST_3SCALE_APP_ID=4d4b20b9
+  - TEST_3SCALE_APP_KEY=ecce202ecc2eb8dc7a499c34a34d5987
+  - secure: B/3oWYZZttrXVM5IFn2evSuWMOQqEOOMol3nEQ2vEDGQ6rdGjpYLFDO/qMDPA8Af6Vt72lOFKkgzivuBp8nSd4S1YRTazilMle4zCJiJxzPs69kRDVXf/Mn8Qr/YrmgVzhblvdNqbKxTGrYyvDziPkliQKtIhqyILm3d6+CqmYw=
+  matrix:
```
Comment (davidor) [inline .travis.yml]: Is this `matrix` required? It wasn't here before.

Comment (avilatusell) [inline .travis.yml]: Changed as global :-)

Comment (davidor) [inline .travis.yml]: :+1:

**Pull request merged**